### DISCUSSION TITLE
-DOQS_MINIMAL_BUILD="OQS_ENABLE_KEM_kyber_512" doesn't build

### DISCUSSION THREAD

**Discussion by azmur8** [open-quantum-safe/liboqs#1981] [Q&A]
Hello, trying to build a library with only one algo, this is current main branch, however it doesn't build:

[1/39] Building C object src/CMakeFiles/oqs.dir/kem/kem.c.o
FAILED: src/CMakeFiles/oqs.dir/kem/kem.c.o
/usr/bin/musl-gcc  -I/home/dr/liboqs/build/include -I/home/dr/liboqs/src -Os -DNDEBUG -fPIC -fvisibility=hidden -Wa,--noexecstack -O3 -fomit-frame-pointer -fdata-sections -ffunction-sections -Wl,--gc-sections -std=gnu11 -MD -MT src/CMakeFiles/oqs.dir/kem/kem.c.o -MF src/CMakeFiles/oqs.dir/kem/kem.c.o.d -o src/CMakeFiles/oqs.dir/kem/kem.c.o -c /home/dr/liboqs/src/kem/kem.c
/home/dr/liboqs/src/kem/kem.c: In function ‘OQS_KEM_new’:
/home/dr/liboqs/src/kem/kem.c:380:24: error: implicit declaration of function ‘OQS_KEM_kyber_512_new’ [-Wimplicit-function-declaration]
  380 |                 return OQS_KEM_kyber_512_new();
      |                        ^~~~~~~~~~~~~~~~~~~~~
/home/dr/liboqs/src/kem/kem.c:380:24: error: returning ‘int’ from a function with return type ‘OQS_KEM *’ makes pointer from integer without a cast [-Wint-conversion]
  380 |                 return OQS_KEM_kyber_512_new();
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
[10/39] Building C object src/common/CMakeFiles/common.dir/sha2/sha2_c.c.o
ninja: build stopped: subcommand failed.

it seems there are few problems here:
1.  OQS_KEM_kyber_512_new is defined in kem_kyber.h which shall be included via kem.h with the following macro:
#ifdef OQS_ENABLE_KEM_KYBER
#include <oqs/kem_kyber.h>
#endif /* OQS_ENABLE_KEM_KYBER */
2. i couldn't find a proper way how to put OQS_ENABLE_KEM_KYBER together with OQS_MINIMAL_BUILD="OQS_ENABLE_KEM_kyber_512", i tried -DOQS_ENABLE_KEM_KYBER=ON but it doesn't compile
3. even if i add inclusion kem_kyber.h of explicitly into kem.c it still doesn't link (looks like kem_kyber_512.c is not compiled, so kem.c is not able to compile OQS_KEM_kyber_512_new)

**Comment by SWilson4** [comment]
Hi @azmur8, thanks for your interest in liboqs! The correct syntax is `-DOQS_MINIMAL_BUILD="KEM_kyber_512"`, without the `OQS_ENABLE_` prefix, as documented in [`CONFIGURE.md`](https://github.com/open-quantum-safe/liboqs/blob/8bf124f1fcc8e73dace94bf2eb8c59441a06b8a8/CONFIGURE.md#OQS_MINIMAL_BUILD).

**Reply by SWilson4** [reply]
I've converted this to a discussion, as there doesn't appear to be a bug in `liboqs`. If the above fix works for you, please consider marking that comment as the answer so that it can help others with similar questions in the future.

**Reply by azmur8** [reply]
argh, ok, thanks. it's different syntax in wiki
https://github.com/open-quantum-safe/liboqs/wiki/Customizing-liboqs/55cfed39e1027dd1d32170e6b91f557571b18d9e

**Reply by SWilson4** [reply]
Thanks for bringing that to our attention! That wiki page should have been updated long ago. I'll edit it so that it points to the current config file (to make sure we stay in sync).

**Reply by SWilson4** [reply]
Hmmm, it looks like the [current revision of the wiki](https://github.com/open-quantum-safe/liboqs/wiki/Customizing-liboqs) does indeed simply point to the CONFIGURE.md file. What led you to the old (55cfed39) revision @azmur8?

**Reply by baentsch** [reply]
> What led you to the old (55cfed39) revision @azmur8?

Good question, indeed: This very problem led me to move (version/release dependent) config options off of the Wiki quite some time ago. Finding/rectifying such stale documentation would help us improve things for others, @azmur8 !

**Reply by azmur8** [reply]
Well, that's the question I won't be able to answer clearly. 
I can't remember exact path I used, to get into that link.
If I query google with "liboqs customizing" the second link gets to that outdated release of wiki.
